### ISSUE TITLE
Command list separate out of DeviceContext

### ISSUE THREAD

**Issue by Qskotlv** [DiligentGraphics/DiligentCore#161]
I think the modern design for DX12 and vulkan is hold the command list in user render code.
How to implement it by myself  in the easiest way ?

**Comment by TheMostDiligent** [comment]
You can record multiple command lists in parallel using deferred contexts, take a look and [tutorail 06](https://github.com/DiligentGraphics/DiligentSamples/tree/master/Tutorials/Tutorial06_Multithreading).

**Comment by TheMostDiligent** [comment]
> What about multi-frame parallel rendering, I'm implementing 3 frames parallelism in Dilligent samples.

You should not do this. Diligent handles multi-frame synchronization where necessary.

**Comment by Qskotlv** [comment]
> > What about multi-frame parallel rendering, I'm implementing 3 frames parallelism in Dilligent samples.
> 
> You should not do this. Diligent handles multi-frame synchronization where necessary.

Cool, where can I find those code ?

**Comment by TheMostDiligent** [comment]
It is hard to tell what code exactly you want to see. Diligent performs a lot of different kinds of synchronization under the hood. You can take a look at implementation of Direct3D12 and Vulkan backends.

**Comment by Qskotlv** [comment]
Ok, thanks.
